manager/controlapi/node.go

```diff
@@ -189,14 +189,18 @@ func (s *Server) UpdateNode(ctx context.Context, request *api.UpdateNodeRequest)
 	var (
 		node   *api.Node
 		demote bool
+		raftID uint64
 	)
+
+	s.raft.MembershipLock.Lock()
+	defer s.raft.MembershipLock.Unlock()
```

Comment:
Let's avoid this if at all possible. Can we expose a RemoveMemberIfAllowed function in raft.go?